data/database.js

```diff
@@ -38,63 +38,114 @@ class Database {
-      } else if (
-        audioType === 'summary' &&
-        fileLocation.summary_audio_location
-      ) {
-        return fileLocation.summary_audio_location;
-      }
-    }
-    return '';
+  async getAudioFileLocation(articleId, type, voice) {
+    logger.info(`getAudioFileLocation for ${articleId}/${type}`);
+    return new Promise(resolve => {
+      AudioFiles.query('item_id')
+        .eq(articleId)
+        .filter('type')
+        .eq(type)
+        .filter('voice')
+        .eq(voice)
+        .filter('codec')
+        .eq('mp3')
+        .exec()
+        .then(function(data) {
+          console.log(data);
+          console.log(JSON.stringify(data));
+          if (data.count) {
+            resolve(data[0].url);
+            if (data.count > 1) {
+              logger.warn('duplicate entries!!!');
+            }
+          } else {
+            resolve('');
+          }
+        });
+    });
   }
 
-  async getMobileFileLocation(articleId) {
-    logger.info(`getMobileFileLocation for ${articleId}`);
-    const fileLocation = await AudioFileLocation.get({ item_id: articleId });
-    if (fileLocation && fileLocation.mobile_audio_location) {
-      return fileLocation.mobile_audio_location;
-    }
-    return '';
+  async getMobileMetadata(articleId) {
+    logger.info(`getMobileMetadata for ${articleId}`);
+    return new Promise(resolve => {
+      AudioFiles.query('item_id')
+        .eq(articleId)
+        .filter('type')
+        .eq('mobile')
+        .exec()
+        .then(function(data) {
+          console.log(data);
+          console.log(JSON.stringify(data));
+          resolve(data);
+        });
+    });
   }
 
-  async storeAudioFileLocation(articleId, audioType, location) {
-    logger.info(
-      `storeAudioFileLocation for ${articleId}/${audioType}: ${location}`
-    );
-    let fileLocation = await AudioFileLocation.get({ item_id: articleId });
-    if (!fileLocation) {
-      fileLocation = new AudioFileLocation({
-        item_id: articleId
-      });
-    }
-    if (audioType === 'full') {
-      fileLocation.full_audio_location = location;
-      fileLocation.full_audio_date = Date.now();
-    } else if (audioType === 'summary') {
-      fileLocation.summary_audio_location = location;
-      fileLocation.summary_audio_date = Date.now();
-    }
-    await fileLocation.save();
+  async storeAudioFileLocation(articleId, type, voice, location) {
+    logger.info(`storeAudioFileLocation for ${articleId}/${type}: ${location}`);
+    let mp3 = new AudioFiles({
+      item_id: articleId,
+      uuid: uuidgen.generate(),
+      voice: voice,
+      codec: 'mp3',
+      bitrate: 40000,
+      samplerate: 16000,
+      type: type,
+      url: location,
+      date: Date.now()
+    });
+    await mp3.save();
+
+    logger.debug('Before opus save');
+    let opus = new AudioFiles({
+      item_id: articleId,
+      uuid: uuidgen.generate(),
+      voice: voice,
+      codec: 'opus',
+      bitrate: 24000,
```

Comment:
This is storing 24000 here for every file but in reality it's up to scout-xcode to decide that and it could inform -ua of the correct rate when the work is done.

Reply:
Yeah, I just hard-coded this because I don't know at this point whether we will want to play around with it and client needs to send it via api, etc.  I suspect someone will ask for some changes and we can tackle it then.